### ISSUE TITLE
Deserialization of accumulated data

### ISSUE THREAD

**Issue by VasilevIvanVladimirovich** [PBernalPolo/triaxialSensorCalibration2019#2]
Hi @PBernalPolo 
I tried to analyze the calibration code and saw that your algorithm consists of two parts;
1) Data accumulation and saving to a file (.dat)
2) Calibration procedure and saving of calibration model

You use the accumulated [file](https://github.com/PBernalPolo/triaxialSensorCalibration2019/blob/cd0205352221f7bad3cf26d74fca69937d36331d/temperatureCalibrationAlgorithm/storedData/data_20190530163305.dat) as calibration input data, can you tell which data correspond to what in order ?

Example of the first line:

`0 17 4 982 913 -17 7 41 -581 -3745 687 -137 0.0`

**Comment by pbernalpolo** [comment]
Hi @VasilevIvanVladimirovich ,

I tried to make it really easy to use, but perhaps I ended up including too much complexity. Let me know how easy is it for you to use after this explanation.

You can find usage examples in:
https://github.com/PBernalPolo/triaxialSensorCalibration2019/blob/master/temperatureCalibrationAlgorithm/temperatureCalibrationAlgorithm.pde
In particular, you can see an example of the usage of the AccelerometerCalibrator in function `calibrate_accelerometer17`:
1. With `AC.set_sensorID( 17 );` we tell AC that the sensor ID is `17`. When it scans through the data file it will select only the measurements associated with that sensor ID.
2. With `AC.set_indexID( 1 );` we tell AC that he will find the sensor ID at the index `1` of each row of data. In the line of your example this would be `17`.
3. With `AC.set_indexTemperature( 11 );` we tell AC to find the temperature at the index `8` of each row of data. In the line of your example that would be `-137`.
4. With `AC.set_indexAngularVelocity( 12 );` we tell AC to find the angular velocity at index `12`. The angular velocity is also important for the accelerometer because it can be used to detect when the sensor is static. AC will discard all measurements with an angular velocity different than 0. In the line of your example it would be `0.0`.
5. With `AC.set_indexMeasurements( 2 , 3 , 4 );` we tell AC to find the accelerometer measurements at indices `2`, `3`, and `4`. In your example this would be `4`, `982`, and `913`.

Looking into `calibrate_gyroscope17` you will check that the indices for gyroscope measurements are `5`, `6`, and `7`, so in your line it is `-17`, `7`, and `41`. And looking into `calibrate_magnetometer17` you will check that the indices for magnetometer measurements are `8`, `9`, and `10`, so in your line it is `-581`, `-3745`, and `687`.

Note that this approach allows different formats for each sensor. For example, if you check `calibrate_accelerometer10`, you will know that the temperature is located in a different position than the one for the sensor with ID 17.

So, to summarize, you don't need to worry about the format to save your data, but you will need to provide the information about where to find each value to the calibration algorithm. :slightly_smiling_face: 

I should also mention the value located at index 0. That one is the same for every sensor; it is just a counter for different sensor orientations. This value was added by the system to keep track of the orientation number. I don't remember if I used it for something else...

**Comment by VasilevIvanVladimirovich** [comment]
Thanks to you, I was able to understand the structure of the accumulated results 👍.
